### DISCUSSION TITLE
About Module of 'InvasionPercolation'

### DISCUSSION THREAD

**Discussion by likeduck** [PMEAL/OpenPNM#2911] [Q&A]
Dear openpnm official, 
I noticed that in the [example](https://openpnm.org/examples/applications/relative_permeability.html) on the official openpnm website, there is an example of predicting the relative permeability of porous media. But I found something worth thinking about (in the figure below).
![捕获](https://github.com/PMEAL/OpenPNM/assets/117707506/d560172b-5015-4176-bb11-beea0af1763c)
The invasion percolation process is described as 'assuming a drainage process, the air(invading/non-wetting phase) will be invading the medium'.  In other words, it is a process of that PNM is filled with water, and air drives the water. So how to make PNM full of oil, with water as a non-wetting phase to drive the oil? I did not find the code to change the original fluid in the PNM in the InvasionPercolation module. Could you please tell me whether this idea is feasible.
Sincerely.

**Comment by jgostick** [comment]
Yes, the IP algorithm just cares about "wetting phase" is defending and "non-wetting phase" is invading.  There is no consideration of viscosity or gravity effects.  These effects could be added, but they are complicated enough that we have never felt comfortable coding a general solution.

**Reply by likeduck** [reply]
I get it, anyway thanks for the work your group has done, I'll keep learning about InvasionPercolation.
